inspirehep/modules/multieditor/actions.py

```diff
@@ -0,0 +1,376 @@
+            else:
+                if not is_primitive_array:
+                    temp_record[key] = [{}]
+                    temp_record = temp_record[key][0]
+    if is_primitive_array:
+        temp_record[path[-1]] = [value]
+    else:
+        temp_record[path[-1]] = value
+    return record
+
+
+def check_value(record, schema, match_type, keypath, update_value, position):
+    """
+    Function that checks the validity of the action condition
+    :param record: the subrecord in which our field of interest resides
+    :param match_type: type of the condition matching
+    :param keypath: path in which the field should be checked
+    :param update_value: value to be matched with the conditions
+    :param position: current recursion level
+    :return:
+    """
+    key = keypath[position]
+    if not schema:
+        return False
+    if not record.get(key):
+        return match_type == 'missing'
+    new_schema = schema_progression(schema, key)
+    temp_record = record[key]
+    if isinstance(temp_record, list):
+        for index, array_record in enumerate(temp_record):
+            if position + 1 == len(keypath):
+                update_value = serialize_value(update_value, new_schema)
+                if update_value == 'error':
+                    return False
+                if match_type == 'exact' and array_record == update_value:
+                    return True
+                elif match_type == 'contains' and update_value.lower() in array_record.lower():
+                    return True
+                elif match_type == 'regex'and re.search(
+                        update_value, array_record):
+                        return True
+            else:
+                if check_value(array_record, new_schema, match_type,
+                               keypath, update_value, position + 1):
+                    return True
+    else:
+        if position + 1 == len(keypath):
+            update_value = serialize_value(update_value, new_schema)
+            if update_value == 'error':
+                return False
+            if match_type == 'exact' and temp_record == update_value:
+                return True
+            elif match_type == 'contains' and update_value.lower() in temp_record.lower():
+                return True
+            elif match_type == 'regex' and re.search(
+                    update_value, temp_record):
+                return True
+        else:
+            return check_value(temp_record, new_schema, match_type,
+                               keypath, update_value, position + 1)
+    return False
+
+
+def serialize_value(value, schema):
+    if schema['type'] == 'array':
+        schema = schema['items']
+    if schema['type'] == 'integer':
+        serialized_value = int(value)
+    elif schema['type'] == 'boolean':
+        if value.lower() == 'true':
+            serialized_value = True
+        elif value.lower() == 'false':
+            serialized_value = False
```

Comment:
The frontend sends a stringified value, im not sure if I can use ```json.loads``` to get the right value, since for instance if a date contains only a year it's still not a number but a string

Reply:
So, we are sending stuff like:
```
{
    "date": 2017,
}
```
?

btw. If we are using some 'serializer' that's not actually json, we should at some point :/
If not, we are inventing our own...